packages/core/workers/src/cpuCount.js

```diff
@@ -1,11 +1,73 @@
-const os = require('os');
+// @flow
+import os from 'os';
+import {execSync} from 'child_process';
 
-module.exports = function() {
-  let cores;
+const exec = (command: string): string => {
   try {
-    cores = require('physical-cpu-count');
-  } catch (err) {
-    cores = os.cpus().length;
+    let stdout = execSync(command, {
+      encoding: 'utf8',
+      // This prevents the command from outputting to the console
+      stdio: [null, null, null],
+    });
+    return stdout.trim();
+  } catch (e) {
+    return '';
   }
-  return cores || 1;
 };
+
+export function detectRealCores(): number {
+  let platform = os.platform();
+  let amount = 0;
+
+  if (platform === 'linux') {
+    let output =
+      exec('nproc') ||
+      exec('lscpu -p | egrep -v "^#" | sort -u -t, -k 2,4 | wc -l');
+
+    amount = parseInt(output, 10);
+  } else if (platform === 'darwin') {
+    amount = parseInt(exec('sysctl -n hw.physicalcpu_max'), 10);
+  } else if (platform === 'windows') {
+    let output = exec('WMIC CPU Get NumberOfCores');
```

Comment:
I copied this command from the `physical-cpu-count` package but apparently it doesn't work in Azure Pipelines.

As I haven't used windows in years I don't really wanna debug this. It's probably not a big deal anyway.